libraries/AP_Mount/AP_Mount_Siyi.cpp

```diff
@@ -0,0 +1,626 @@
+        _last_req_current_angle_rad_ms = now_ms;
+    }
+
+    // update based on mount mode
+    switch (get_mode()) {
+        // move mount to a "retracted" position.  To-Do: remove support and replace with a relaxed mode?
+        case MAV_MOUNT_MODE_RETRACT: {
+            const Vector3f &angle_bf_target = _params.retract_angles.get();
+            send_target_angles(ToRad(angle_bf_target.y), ToRad(angle_bf_target.z), false);
+            break;
+        }
+
+        // move mount to a neutral position, typically pointing forward
+        case MAV_MOUNT_MODE_NEUTRAL: {
+            const Vector3f &angle_bf_target = _params.neutral_angles.get();
+            send_target_angles(ToRad(angle_bf_target.y), ToRad(angle_bf_target.z), false);
+            break;
+        }
+
+        // point to the angles given by a mavlink message
+        case MAV_MOUNT_MODE_MAVLINK_TARGETING:
+            switch (mavt_target.target_type) {
+            case MountTargetType::ANGLE:
+                send_target_angles(mavt_target.angle_rad.pitch, mavt_target.angle_rad.yaw, mavt_target.angle_rad.yaw_is_ef);
+                break;
+            case MountTargetType::RATE:
+                send_target_rates(mavt_target.rate_rads.pitch, mavt_target.rate_rads.yaw, mavt_target.rate_rads.yaw_is_ef);
+                break;
+            }
+            break;
+
+        // RC radio manual angle control, but with stabilization from the AHRS
+        case MAV_MOUNT_MODE_RC_TARGETING: {
+            // update targets using pilot's rc inputs
+            MountTarget rc_target {};
+            if (get_rc_rate_target(rc_target)) {
+                send_target_rates(rc_target.pitch, rc_target.yaw, rc_target.yaw_is_ef);
+            } else if (get_rc_angle_target(rc_target)) {
+                send_target_angles(rc_target.pitch, rc_target.yaw, rc_target.yaw_is_ef);
+            }
+            break;
+        }
+
+        // point mount to a GPS point given by the mission planner
+        case MAV_MOUNT_MODE_GPS_POINT: {
+            MountTarget angle_target_rad {};
+            if (get_angle_target_to_roi(angle_target_rad)) {
+                send_target_angles(angle_target_rad.pitch, angle_target_rad.yaw, angle_target_rad.yaw_is_ef);
+            }
+            break;
+        }
+
+        case MAV_MOUNT_MODE_HOME_LOCATION: {
+            MountTarget angle_target_rad {};
+            if (get_angle_target_to_home(angle_target_rad)) {
+                send_target_angles(angle_target_rad.pitch, angle_target_rad.yaw, angle_target_rad.yaw_is_ef);
+            }
+            break;
+        }
+
+        case MAV_MOUNT_MODE_SYSID_TARGET:{
+            MountTarget angle_target_rad {};
+            if (get_angle_target_to_sysid(angle_target_rad)) {
+                send_target_angles(angle_target_rad.pitch, angle_target_rad.yaw, angle_target_rad.yaw_is_ef);
+            }
+            break;
+        }
+
+        default:
+            // we do not know this mode so do nothing
```

Comment:
Getting here should probably be a flow-of-control internal error, but meh.

Reply:
ok, added an internal error